deepmedic/routines/training.py

```diff
@@ -24,382 +24,437 @@
-                                savePredictedSegmAndProbsDict,
-                                listOfFilepathsToEachChannelOfEachPatientValidation,
-                                listOfFilepathsToGtLabelsOfEachPatientValidationOnSamplesAndDsc,
-                                listOfFilepathsToRoiMaskOfEachPatientValidation,
-                                namesForSavingSegmAndProbs = namesForSavingSegmAndProbs,
-                                suffixForSegmAndProbsDict = suffixForSegmAndProbsDict,
-                                # Hyper parameters
-                                batchsize = batchsize_val_whole,
-                                
-                                #----Preprocessing------
-                                pad_input_imgs = pad_input_imgs,
-                                
-                                #--------For FM visualisation---------
-                                saveIndividualFmImagesForVisualisation = saveIndividualFmImagesForVisualisation,
-                                saveMultidimensionalImageWithAllFms = saveMultidimensionalImageWithAllFms,
-                                indicesOfFmsToVisualisePerPathwayTypeAndPerLayer = indicesOfFmsToVisualisePerPathwayTypeAndPerLayer,
-                                namesForSavingFms = namesForSavingFms
-                                )
+                log.print3(
+                    "***Starting validation with Full Inference / Segmentation on validation subjects for Epoch #" + str(
+                        epoch) + "...***")
+
+                metrics_dict_list = inferenceWholeVolumes(
+                    sessionTf,
+                    cnn3d,
+                    log,
+                    "val",
+                    savePredictedSegmAndProbsDict,
+                    listOfFilepathsToEachChannelOfEachPatientValidation,
+                    listOfFilepathsToGtLabelsOfEachPatientValidationOnSamplesAndDsc,
+                    listOfFilepathsToRoiMaskOfEachPatientValidation,
+                    namesForSavingSegmAndProbs=namesForSavingSegmAndProbs,
+                    suffixForSegmAndProbsDict=suffixForSegmAndProbsDict,
+                    # Hyper parameters
+                    batchsize=batchsize_val_whole,
+
+                    # ----Preprocessing------
+                    pad_input_imgs=pad_input_imgs,
+
+                    # --------For FM visualisation---------
+                    saveIndividualFmImagesForVisualisation=saveIndividualFmImagesForVisualisation,
+                    saveMultidimensionalImageWithAllFms=saveMultidimensionalImageWithAllFms,
+                    indicesOfFmsToVisualisePerPathwayTypeAndPerLayer=indicesOfFmsToVisualisePerPathwayTypeAndPerLayer,
+                    namesForSavingFms=namesForSavingFms
+                )
+
+                acc_monitor_for_ep_val.reportDSCWholeSegmentation(metrics_dict_list)
+
+            del acc_monitor_for_ep_train
+            del acc_monitor_for_ep_val
```

Comment:
I think we can safely delete these. The garbage collector will do the job anyway.